packages/serverpod/lib/src/server/websocket_request_handlers/method_websocket_request_handler.dart

```diff
@@ -391,7 +413,44 @@ class _MethodStreamManager {
     WebSocket webSocket,
     Session session,
     SerializationManager serializationManager,
+    _RequiredAuthentication? requiredAuthentication,
   ) {
+    late void Function(SerializableModel) authenticationRevokedCallback;
+
+    if (requiredAuthentication != null) {
+      // If authentication was required, a listener is registered that closes
+      // the stream if the authentication token is revoked.
+      authenticationRevokedCallback = (event) async {
+        var authenticationRevoked = switch (event) {
+          RevokedAuthenticationUser _ => true,
+          RevokedAuthenticationAuthId revokedAuthId =>
+            revokedAuthId.authId == requiredAuthentication.authInfo.authId,
+          RevokedAuthenticationScope revokedScopes => revokedScopes.scopes.any(
+              (s) => requiredAuthentication.requiredScopes
+                  .map((s) => s.name)
+                  .contains(s),
+            ),
+          _ => false,
+        };
+
+        if (authenticationRevoked) {
+          await closeStream(
+            endpoint: message.endpoint,
+            method: message.method,
+            connectionId: message.connectionId,
+            reason: CloseReason.error,
+          );
+        }
+      };
+
+      session.messages.addListener(
+        MessageCentralServerpodChannels.revokedAuthentication(
+          requiredAuthentication.authInfo.userId,
+        ),
+        authenticationRevokedCallback,
```

Comment:
What happens if the `if` was not entered, will `authenticationRevokedCallback` be uninitialized then? If so, wouldn't that risk throwing later if some code tried to check for it?

Reply:
That is true, in this case we have the whole world so we know we are safe. But it is brittle. I can switch it up a bit to let the function in itself determine if we should remove it or not.

Reply:
Check out the new version now.

Reply:
Looks good! 👍